hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapFetchEntriesOperation.java

```diff
@@ -60,6 +60,13 @@ public Object getResponse() {
         return response;
     }
 
+    // HD scan via operation supposed to be thread-safe, no need to control concurrency.
+    @Override
+    protected void assertNativeMapOnPartitionThread() {
+        assert getPartitionId() != GENERIC_PARTITION_ID
+            : "Native memory backed map operations are not allowed to run on GENERIC_PARTITION_ID";
+    }
+
```

Comment:
Why don't we just remove this method and don't rely on the superclass?

Reply:
Sorry, didn't get you. This method is called in `MapOperation#beforeRun` method, which is operation lifecycle method.
If we'd remove it, this operation can't get access to HD memory data.